### DISCUSSION TITLE
Social Login and email smart addresses help with Appkit web3modal latest and ethers 5 as per latest documentation on appkit

### DISCUSSION THREAD

**Discussion by CheyneWeb3** [WalletConnect/walletconnect-monorepo#4622] [Developer Support] [answered]
Im using app.tsx file with Vite.js ill provide file to assist with question .

question is regarding the social accounts i cant seem to get them to appear in the modal and on building the auth: returns errors.

ive followed the documentation and tested the labs version i have everything running well now with latest package .

but i cannot get it to run or build if auth set on file

provided the app.tsx file
[App.tsx.txt](https://github.com/user-attachments/files/15861337/App.tsx.txt)
[package.json.txt](https://github.com/user-attachments/files/15861343/package.json.txt)

any assistance greatly accepted

site is https://thatdamndawg.com

**Comment by KingSimpa69** [comment, marked as answer]
Hey brother, I'm in the same boat as you. I've been a long time user of ethers v5 because I couldn't get event listeners to work correctly when v6 first came out. But I guess It's time to get with the program and make the upgrade. Don't want to be left in the dust. Onboarding with email and also smart wallets is gonna be the future if we want mass adoption.

web3modal for v6 ethers has the email connector config:
[https://github.com/WalletConnect/web3modal/blob/main/packages/ethers/src/utils/defaultConfig.ts](https://github.com/WalletConnect/web3modal/blob/main/packages/ethers/src/utils/defaultConfig.ts)

web3modal v5 doesn't:
[https://github.com/WalletConnect/web3modal/blob/main/packages/ethers5/src/utils/defaultConfig.ts](https://github.com/WalletConnect/web3modal/blob/main/packages/ethers5/src/utils/defaultConfig.ts)

There might be support for it down the road, but if you're currently using the ethers v5 version of web3modal It doesn't appear to be supported.

**Reply by CheyneWeb3** [reply]
yes thankyou i tried the update to ethers6 and the social logins worked correctly . thanks for answering the discuusion and for the links to use i really apprieciate the dev help .

**Comment by KingSimpa69** [comment]
Can confirm, upgrading to v6 solved it for me...
![Yc3pt6S](https://github.com/WalletConnect/walletconnect-monorepo/assets/114375407/40c7569d-5d62-4e41-9d6e-27f9eef63d1a)

**Reply by CheyneWeb3** [reply]
awesome thanks everybody
![telegram-cloud-photo-size-5-6316704386612576723-y](https://github.com/WalletConnect/walletconnect-monorepo/assets/104659633/7250d8bc-4172-48ed-a604-b42c174e561f)
